languages/dart/keywords/abstract.md

```diff
@@ -0,0 +1,102 @@
+# Abstract
+
+The keyword `abstract` is used to define an *abstract class* - a [class][reference-class] that can not be instantiated 
+directly, which means you can not create an [object][reference-object] from it. Conversely, a class that can be 
+instantiated directly is called a *concrete class*.
+
+Abstract types are [interfaces][reference-interface] that hide the implementation details and show only functionality to
+the user. They let you focus on **what** the object does instead of **how** it does it.
+
+```dart
+// This class is declared abstract and thus can't be instantiated.
+abstract class CoffeeMachine {
+  //Body of abstract class
+}
+
+// Abstract classes can't be instantiated.
+CoffeeMachine coffeeMachine = CoffeeMachine();
+```
+
+An abstract class can have constructors (including [factories][reference-factory]), [fields][reference-field], 
+[methods][reference-method] as well as [static][reference-static] [member][reference-member].
+Among these only fields and methods can be abstract everything else, if defined, must have a concrete implementation. 
+Abstract methods define the `class` behaviour. They can only exist within an `abstract class` and are delimiting by a 
+semicolon (`;`).
+
+```dart
+abstract class CoffeeMachine {
+  // Abstract method
+  void makeEspresso(); 
+  
+  // Normal method
+  void makeCoffee() { 
+    print("Making coffee");
+  }
+}
+```
+
+Since abstract classes can have constructors and constructors in Dart can omit their body, a constructor without a body 
+does not denote that it is abstract.
+
+Constructors in Dart are always concrete.
+
+```dart
+// Concrete class
+class MyClass1 {
+  // Constructor without a body
+  MyClass1.one();
+  // Constructor with an empty body
+  MyClass1.two(){}
+}
+
+// Abstract class
+abstract class MyClass2 {
+  // Constructor without a body
+  MyClass2.one();
+  // Constructor with an empty body
+  MyClass2.two(){}
+}
+```
+
+When concrete classes [extend][reference-extend] the abstract class, they have to [override][reference-override]
+every abstract method but it is not mandatory to override normal method.
```

Comment:
I would suggest adding to the example SpecialCoffeeMachine using makeCoffee in some way. To further highlight that the normal method did not need to be overridden.